hazelcast-client/src/test/java/com/hazelcast/client/txn/ClientTxnTest.java

```diff
@@ -144,8 +146,27 @@ public void memberRemoved(MembershipEvent membershipEvent) {
         assertOpenEventually(txnRollbackLatch);
         assertOpenEventually(memberRemovedLatch);
 
-        final IQueue<Object> q = hz.getQueue(queueName);
+        final IQueue<Object> q = client.getQueue(queueName);
         assertNull(q.poll());
         assertEquals(0, q.size());
     }
+
+    @Test
+    public void testRollbackOnTimeout() {
+        String name = randomString();
+        IQueue<Object> queue = client.getQueue(name);
+        queue.offer(randomString());
+
+        TransactionOptions options = new TransactionOptions().setTimeout(2, TimeUnit.SECONDS);
+        TransactionContext context = client.newTransactionContext(options);
+        context.beginTransaction();
+        try {
+            context.getQueue(name).take();
+            sleepAtLeastSeconds(3);
+            context.commitTransaction();
```

Comment:
May be we can put fail() check after this line to make sure it is not executed for sure. Similarly for the XA test.